context/Auth.js

```diff
@@ -0,0 +1,92 @@
+import { useRouter, useSegments } from "expo-router";
+import { onAuthStateChanged } from "firebase/auth";
+import { createContext, useContext, useEffect, useState } from "react";
+import { auth } from "~/firebaseConfig";
+
+const AuthContext = createContext({
+  signIn() {
+    return;
+  },
+  signOut() {
+    return;
+  },
+  setUser() {
+    return;
+  },
+  user: null,
+  teamId: null,
+});
+
+export function currentAuthContext() {
+  return useContext(AuthContext);
+}
+
+function useProtectedRoute(user) {
+  const segments = useSegments();
+  const router = useRouter();
+
+  useEffect(() => {
+    const inAuthGroup = segments.at(0) === "(auth)";
+
+    if (!user && !inAuthGroup) {
+      router.replace("/signin");
+    } else if (user && inAuthGroup) {
+      router.replace("/");
+    }
+  }, [user, segments]);
+}
+
+export const AuthProvider = ({ children }) => {
+  const [user, setUser] = useState(null);
+  const [teamId, setTeamId] = useState("1"); // TODO: Set this properly instead of hardcoding team
+
+  useProtectedRoute(user);
+
+  useEffect(() => {
+    //if this code is not in here, it'll run for infinite times
+    onAuthStateChanged(auth, (user) => {
+      console.log("user changed");
+
+      // yarn test
+      // If you sign out, reload app to sign back in as test user
+      // TODO: After setting users properly, set test user too
+      if (process.env.EXPO_PUBLIC_TEST_UID) {
+        setUser({
+          name: process.env.EXPO_PUBLIC_TEST_NAME,
+          email: process.env.EXPO_PUBLIC_TEST_EMAIL,
+          uid: process.env.EXPO_PUBLIC_TEST_UID,
+        });
+      }
+      if (user) {
+        setUser({
+          name: user.displayName ?? "Error (name)",
```

Comment:
I think there has been a lot of confusion around this but my opinion is there shouldn't be a display name for a user id, but only for a user id inside a team. This allows a logged in user to have multiple display name for each teams.

Reply:
changed to be just uid in there